### PR TITLE
Log important statistics to help debug memory issues

### PR DESCRIPTION
**Pull request by coreyfloyd**
Logging on launch/background and when an article is pushed

Also adding in:
https://phabricator.wikimedia.org/T141629

For sending log files with our support messages

### CODE REVIEW

File: Wikipedia/Code/WMFAppViewController.m [hunk 1]
```diff
@@ -275,6 +280,21 @@ - (void)didReceiveMemoryWarning {
     [[[SessionSingleton sharedInstance] dataStore] clearMemoryCache];
 }
 
+#pragma mark - Logging
+
+- (void)logImportantStatistics {
+    NSUInteger historyCount       = [self.session.dataStore.userDataStore.historyList countOfEntries];
+    NSUInteger saveCount          = [self.session.dataStore.userDataStore.historyList countOfEntries];
```
Comment (montehurd) [inline Wikipedia/Code/WMFAppViewController.m]: you have "historyList" twice

File: Wikipedia/Code/WMFHelpViewController.m [hunk 2]
```diff
@@ -62,6 +63,10 @@ - (void)sendEmail {
         [vc setSubject:[WMFSettingsEmailSubject stringByAppendingString:[WikipediaAppUtils versionedUserAgent]]];
         [vc setToRecipients:@[WMFSettingsEmailAddress]];
         [vc setMessageBody:[NSString stringWithFormat:@"\n\n\n\nVersion: %@", [WikipediaAppUtils versionedUserAgent]] isHTML:NO];
+        NSData* data = [[DDLog wmf_currentLogFile] dataUsingEncoding:NSUTF8StringEncoding];
+        if (data) {
+            [vc addAttachmentData:data mimeType:@"text/plain" fileName:@"Log Data.txt"];
```
Comment (montehurd) [inline Wikipedia/Code/WMFHelpViewController.m]: maybe name "Item Counts.txt" so we're less tempted to use this a catch-all for log info?

Comment (coreyfloyd) [inline Wikipedia/Code/WMFHelpViewController.m]: @montehurd it is all the log info, not just the counts

Comment (coreyfloyd) [inline Wikipedia/Code/WMFHelpViewController.m]: @montehurd sorry this was a second ticket just putting on the same PR:

https://phabricator.wikimedia.org/T141629

File: Wikipedia/Code/WMFAppViewController.m (hunk 1 above)
Comment (coreyfloyd) [inline Wikipedia/Code/WMFAppViewController.m]: Fixed

**Comment by montehurd** [comment]
@coreyfloyd oh gotcha. should we get the @"Log Data.txt" string from a common var then? (i thought it was newly added)

**Comment by coreyfloyd** [comment]
@montehurd it only used in one place for the title of the file we send to ourselves, so I don't think we need to make a variable out of it

**Comment by joewalsh** [comment]
:+1:

File: Wikipedia/Code/WMFExploreViewController.m [hunk 3]
```diff
@@ -154,6 +154,10 @@ - (WMFExploreSectionControllerCache*)sectionControllerCache {
     return _sectionControllerCache;
 }
 
+- (NSUInteger)numberOfSectionsInExploreFeed{
```
Comment (montehurd) [inline Wikipedia/Code/WMFExploreViewController.m]: Oh were weren't we gonna standardize on signed integers?

**Comment by montehurd** [comment]
👍

File: Wikipedia/Code/WMFExploreViewController.m (hunk 3 above)
Comment (coreyfloyd) [inline Wikipedia/Code/WMFExploreViewController.m]: Did we decide to do that in Obj-C as well?

Comment (montehurd) [inline Wikipedia/Code/WMFExploreViewController.m]: I think we prob should? To ease future Swift inter-op and/or conversion to Swift...

**Pull request merged**